internal/workerpool/pool.go

```diff
@@ -39,8 +40,11 @@ func Run(ctx context.Context, limit int, nextTask func(context.Context) (Task, e
 	var retErr error
 	setErr := func(err error) {
 		mu.Lock()
-		retErr = err
+		if retErr == nil {
+			retErr = fmt.Errorf("workerpool: %v", err)
```

Comment:
I don't think it's valuable to wrap here -- you lose too much information for too little gain. (workerpool is an internal package, so identifying it doesn't really help much.) Just use err.

Reply:
Okay, reverted it.